controllers/yawol-controller/loadbalancerset/helper.go

```diff
@@ -60,41 +61,60 @@ func areRelevantConditionsMet(machine *yawolv1beta1.LoadBalancerMachine, expirat
 	return true, nil
 }
 
-func getDeletionCondition(machine *yawolv1beta1.LoadBalancerMachine) (bool, *corev1.NodeCondition) {
+func findDeletionCondition(machine *yawolv1beta1.LoadBalancerMachine) *corev1.NodeCondition {
 	if machine.Status.Conditions == nil {
-		return false, nil
+		return nil
 	}
 	conditions := *machine.Status.Conditions
 	for i := range conditions {
 		if conditions[i].Type == helper.DeletionMarkerCondition {
-			return true, &conditions[i]
+			return &conditions[i]
 		}
 	}
-	return false, nil
+	return nil
 }
 
-func setDeletionCondition(machine *yawolv1beta1.LoadBalancerMachine, status corev1.ConditionStatus, reason, message string) {
+func setDeletionCondition(machine *yawolv1beta1.LoadBalancerMachine, newCondition corev1.NodeCondition) {
 	if machine.Status.Conditions == nil {
 		machine.Status.Conditions = &[]corev1.NodeCondition{}
 	}
-	var cond *corev1.NodeCondition
-	conditions := *machine.Status.Conditions
-	if ok, foundCondition := getDeletionCondition(machine); ok {
-		cond = foundCondition
-	} else {
-		// TODO: do we need to sort the slice again?
-		conditions = append(conditions, corev1.NodeCondition{
-			Type: helper.DeletionMarkerCondition,
-		})
-		cond = &(conditions)[len(conditions)-1]
+
+	newCondition.LastHeartbeatTime = metav1.Now()
```

Comment:
We don't need to set `lastHeartbeatTime`.
Before moving to leases, this field was originally used for the kubelet's heartbeats.